cmd/tiller/release_server.go

```diff
@@ -678,7 +679,10 @@ func (s *releaseServer) renderResources(ch *chart.Chart, values chartutil.Values
 	notes := ""
 	for k, v := range files {
 		if strings.HasSuffix(k, notesFileSuffix) {
-			notes = v
+			// Only apply the notes if it belongs to the parent chart
+			if k == filepath.Join(ch.Metadata.Name, notesFileSuffix) {
```

Comment:
I think you were right, it is `$NAME/templates/NOTES.txt` when running tiller for real, but in the tests it's `$NAME/NOTES.txt`. Not sure why it was like that in the tests, but I should have checked it when running tiller at the time D: